### PR TITLE
Refactored the example to a higher level of best practices

### PR DESCRIPTION
**Pull request by DsanchezPerez**
Looking to improve the level of code quality.

I am not aware of any issues for this project but from reviewing the code, it looked like some adjustments would help present the article with Best Practices implementations.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetDocs?pullRequest=805) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetDocs?pullRequest=805) before we can accept your contribution.<br/><br/>:x: DsanchezPerez [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetDocs?pullRequest=805)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetDocs?pullRequest=805) it.</sub>

File: aspnet/signalr/overview/advanced/dependency-injection/samples/sample3.cs [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
     public void DoSomething()
     {
-        _logger.LogMessage("DoSomething");
+        try
+        {
+            _logger.LogMessage($"2 + 2 = {2 + 2}");
+        }
+        catch(Exception ex)
+        {
+            _logger.LogError(new EventId(50100, "SignalR_IOC_DI"), ex, ex.Message);
```
Comment (BrennanConroy) [inline aspnet/signalr/overview/advanced/dependency-injection/samples/sample3.cs:20]: These docs are for ASP.NET SignalR and not ASP.NET Core SignalR. This means that there are a lot of different APIs between the two worlds, e.g. `EventId` is an ASP.NET Core concept, so should not be used here.

File: aspnet/signalr/overview/advanced/dependency-injection/samples/sample3.cs [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
     }
 
     public void DoSomething()
     {
-        _logger.LogMessage("DoSomething");
+        try
+        {
+            _logger.LogMessage($"2 + 2 = {2 + 2}");
+        }
+        catch(Exception ex)
```
Comment (BrennanConroy) [inline aspnet/signalr/overview/advanced/dependency-injection/samples/sample3.cs:18]: What is throwing? This seems unnecessary

File: aspnet/signalr/overview/advanced/dependency-injection/samples/sample3.cs [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
 
     // Inject ILogger into the object.
     public SomeComponent(ILogger logger)
     {
-        if (logger == null)
-        {
-            throw new NullReferenceException("logger");
-        }
-        _logger = logger;
+        _logger = logger ?? throw new NullReferenceException(nameof("logger"));
```
Comment (BrennanConroy) [inline aspnet/signalr/overview/advanced/dependency-injection/samples/sample3.cs:9]:
```suggestion
        _logger = logger ?? throw new NullReferenceException(nameof(logger));
```

**Pull request closed without merging**